tests/unit/test_multicast_listeners_table.cpp

```diff
@@ -0,0 +1,211 @@
+ *  modification, are permitted provided that the following conditions are met:
+ *  1. Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *  2. Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *  3. Neither the name of the copyright holder nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "openthread-core-config.h"
+
+#if OPENTHREAD_FTD && OPENTHREAD_CONFIG_BACKBONE_ROUTER_ENABLE
+
+#include "test_platform.h"
+
+#include <openthread/config.h>
+#include <openthread/ip6.h>
+
+#include "test_util.h"
+#include "backbone_router/multicast_listeners_table.hpp"
+#include "common/code_utils.hpp"
+#include "common/instance.hpp"
+
+namespace ot {
+
+static ot::Instance *sInstance;
+
+using namespace ot::BackboneRouter;
+
+static const otIp6Address MA201 = {
+    {{0xff, 0x02, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01}}};
+static const otIp6Address MA301 = {
+    {{0xff, 0x03, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01}}};
+static const otIp6Address MA401 = {
+    {{0xff, 0x04, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01}}};
+static const otIp6Address MA501 = {
+    {{0xff, 0x05, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01}}};
+
+uint32_t sNow;
+
+uint32_t testTimerAlarmGetNow(void)
+{
+    return sNow;
+}
+
+void InitTestTimer(void)
+{
+    g_testPlatAlarmGetNow = testTimerAlarmGetNow;
+}
+
+void testMulticastListenersTableAPIs(Instance *aInstance);
+
+void TestMulticastListenersTable(void)
+{
+    InitTestTimer();
+
+    sInstance = testInitInstance();
+    VerifyOrQuit(sInstance != nullptr, "Null OpenThread instance");
+
+    MulticastListenersTable table = sInstance->Get<MulticastListenersTable>();
```

Comment:
This creates a copy of table. If it is not intentional, I suggest using a reference instead?

(may want to consider declaring `MulticastListenersTable` as `NonCopyable`?)